source/analysis/AMAdditionAB.cpp

```diff
@@ -104,6 +104,15 @@ bool AMAdditionAB::axisValues(int axisNumber, int startIndex, int endIndex, doub
 void AMAdditionAB::onInputSourceValuesChanged(const AMnDIndex& start, const AMnDIndex& end)
 {
 	cacheUpdateRequired_ = true;
+
+	AMnDIndex scanStart = start;
+	AMnDIndex scanEnd = end;
+	scanStart.setRank(start.rank()-1);
+	scanEnd.setRank(end.rank()-1);
+
+	if (scanStart == scanEnd)
+		dirtyIndices_ << start;
+
```

Comment:
I feel like I'm missing something here with respect to the dirtyIndices_ only being set when the source values change one at a time. What happens when a range of values changes and start != end?
(this applies to all the onInputSourceValuesChanged(...)  calls)